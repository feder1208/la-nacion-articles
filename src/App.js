import "./App.css";
import React, { useState, useEffect } from "react";
import { Loading } from "./components/Loading";
// import { AccumulatedGrid } from "./components/accumulatedGrid";
import { AccumulatedGrid } from "./components/accumulatedGrid";
import './components/index.css'
// import {fetchData} from "./services/fetchData"

const url = "https://api-test-ln.herokuapp.com/articles";

function App() {
  const [articles, setArticles] = useState([]);
  // const [loading, setLoading] = useState(true);

  const fetchData = async () => {
    // setLoading(true);
    const response = await fetch(url);
    const data = await response.json();
    const { articles = [] } = data || {};
    setArticles(articles);
    // console.log(articles, "en app.js");
    // console.log(articles.articles[0].headlines.basic);
    // const mapTitles = articles.articles.map((article) => {
    //   return article.headlines.basic;
    // });
    // console.log(mapTitles, "titulos en app");
  };

  useEffect(() => {
    fetchData();
  }, []);

  // if (loading) {
  //   return (
  //     <main>
  //       <Loading />
  //     </main>
  //   );
  // }

  return <AccumulatedGrid articles={articles} />;
}

export default App;
