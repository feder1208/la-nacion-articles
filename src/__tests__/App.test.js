import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

test("App component test", () => {
    const {container, debug} = render(<App />)
    // debug();
  expect(container).toMatchSnapshot();
});
