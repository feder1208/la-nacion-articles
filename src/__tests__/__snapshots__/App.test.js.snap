// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test 1`] = `
<div>
  <main>
    <div>
      <h1>
        loading...
      </h1>
    </div>
  </main>
</div>
`;
